File: frontend/src/types/task.ts
export type TaskStatus = "pending" | "running" | "completed" | "failed";

export interface Task {
  id: string;
  type: string;
  status: TaskStatus;
  progress: number;
  message: string | null;
}

export interface TaskProgressEvent {
  taskId: string;
  status: TaskStatus;
  progress: number;
  message: string | null;
}

File: backend/app/schemas/task.py
from enum import StrEnum

from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel


class TaskStatus(StrEnum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"


class TaskRead(BaseModel):
    id: str
    type: str
    status: TaskStatus
    progress: int
    message: str | None = None


class TaskProgressEvent(BaseModel):
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    task_id: str
    status: TaskStatus
    progress: int
    message: str | None = None
